rust/src/lib.rs

```diff
@@ -84,22 +89,23 @@ pub extern "C" fn context_create_with_schema(
     schema_ptr: *mut Schema,
     error_buffer: *mut *mut c_char,
 ) -> *mut TantivyContext {
-    let schema = match assert_pointer(schema_ptr, error_buffer) {
-        Some(value) => value.clone(),
-        None => return ptr::null_mut(),
-    };
-
-    let path = match assert_string(path_ptr, error_buffer) {
-        Some(value) => value,
-        None => return ptr::null_mut(),
+    let result = || -> Result<*mut TantivyContext, TantivyGoError> {
+        let schema = assert_pointer(schema_ptr)?.clone();
+        let path = assert_string(path_ptr)?;
+        create_context_with_schema(schema, path)
```

Comment:
I'd do a separate match for each line here -- but it seems you've done it this way on purpose?